client/shared/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -234,7 +234,7 @@ object Client {
     * @param app the [[HttpApp]] to respond to requests to this client
     */
   def fromHttpApp[F[_]](app: HttpApp[F])(implicit F: Async[F]): Client[F] = {
```

Comment:
Any chance we can change F to `F[_]: MonadCancelThrow: Ref.Make` here?

Reply:
Not crazy about `Ref.Make` (what's the motivation?) but maybe this could become `Concurrent`.

Reply:
Motivation in general: `Ref.Make` works for `F`s that are Sync but not Concurrent, but doesn't require the full powers of `Sync`. In this case `Concurrent` probably works better

Reply:
added Concurrent and MonadCancelThrow